lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -4932,6 +4930,41 @@ namespace Js
         }
     }
 
+    bool JavascriptOperators::CanShortcutOnUnknownPropertyName(RecyclableObject *instance)
+    {
+        if (!CanShortcutInstanceOnUnknownPropertyName(instance))
+        {
+            return false;
+        }
+        return CanShortcutPrototypeChainOnUnknownPropertyName(instance->GetPrototype());
+    }
+
+    bool JavascriptOperators::CanShortcutInstanceOnUnknownPropertyName(RecyclableObject *instance)
+    {
+        if (!instance->HasDeferredTypeHandler())
```

Comment:
E.g. an instance that is a Proxy will not have a deferred type handler.  So it'll come here and return true instead of returning false below?

---
In reply to: [59070623](https://github.com/Microsoft/ChakraCore/pull/723#discussion_r59070623) [](ancestors = 59070623)